### PR TITLE
Temporarily deploy webhooks with failurePolicy set to Ignore

### PR DESCRIPTION
**Pull request by mmirecki**
Deploying webhooks with a non Ignore failure policy can negatively affect the stability of the cluster if the deployment backing it fails. This PR causes the webhooks to be created with an Ignore failure policy, and changes it to the desired value once the gatekeeper controller deployment is fully operational

### CODE REVIEW

File: controllers/gatekeeper_controller.go [hunk 1]
```diff
@@ -194,69 +199,137 @@ func (r *GatekeeperReconciler) SetupWithManager(mgr ctrl.Manager) error {
...
-		// Handle special cases in switch below.
-		switch {
-		case a == NamespaceFile && !r.isOpenShift():
-			// Ignore the namespace resource on Kubernetes as we default to use
-			// the same namespace as the operator, which by definition is
-			// already created as a result of executing this code.
-			continue
-		case a == RoleFile && r.isOpenShift():
-			a = openshiftAssetsDir + a
+	// Checking for deployment before deploying assets to avoid cert rotator erors
```
Comment (font) [inline controllers/gatekeeper_controller.go:215]: s/erors/errors/

File: controllers/merge/merge.go [hunk 2]
```diff
@@ -39,6 +39,8 @@ func RetainClusterObjectFields(desiredObj, clusterObj *unstructured.Unstructured
 		fallthrough
 	case util.MutatingWebhookConfigurationKind:
 		return retainWebhookConfigurationFields(desiredObj, clusterObj)
+	case util.SecretKind:
```
Comment (font) [inline controllers/merge/merge.go:42]: Doh, good catch!

File: test/e2e/gatekeeper_controller.go [hunk 3]
```diff
@@ -46,9 +46,9 @@ import (
 
 const (
 	// The length of time between polls.
-	pollInterval = 50 * time.Millisecond
+	pollInterval = time.Second
```
Comment (font) [inline test/e2e/gatekeeper_controller.go:49]: This is OK for now, but I'd prefer to leave a low default for local testing and allow it to be configurable via command line so we can set a higher value when running in CI.

File: test/e2e/gatekeeper_controller.go [hunk 4]
```diff
@@ -46,9 +46,9 @@ import (
...
 	// How long to try before giving up.
-	waitTimeout = 30 * time.Second
+	waitTimeout = 60 * time.Second
```
Comment (font) [inline test/e2e/gatekeeper_controller.go:51]: Same comment as above.

**Review by font — COMMENTED** [review]
Amazing @mmirecki! I made a couple comments and one minor nit. I re-ran the tests to verify it passes consistently, but ran into an issue that I've fixed in #151. Please rebase so the test passes again.

File: controllers/gatekeeper_controller.go (hunk 1 above)
Comment (mmirecki) [reply controllers/gatekeeper_controller.go:215]: Done

File: test/e2e/gatekeeper_controller.go (hunk 3 above)
Comment (mmirecki) [reply test/e2e/gatekeeper_controller.go]: Done

File: test/e2e/gatekeeper_controller.go (hunk 4 above)
Comment (mmirecki) [reply test/e2e/gatekeeper_controller.go]: Done

**Comment by mmirecki** [comment]
> Amazing @mmirecki! I made a couple comments and one minor nit. I re-ran the tests to verify it passes consistently, but ran into an issue that I've fixed in #151. Please rebase so the test passes again.

Rebased

**Review by font — APPROVED** [review]

File: controllers/merge/merge.go (hunk 2 above)
Comment (font) [reply controllers/merge/merge.go:42]: @mmirecki I'm actually thinking that this change may be all that was needed to make #149 work successfully. WDYT?

**Pull request merged**